nexus/reconfigurator/planning/src/planner.rs

```diff
@@ -1572,11 +1580,40 @@ impl<'a> Planner<'a> {
     /// case, we have no choice but to reconcile with the fact that the zone is
     /// now gone.
     fn can_zone_be_shut_down_safely(&self, zone: &BlueprintZoneConfig) -> bool {
-        // TODO-cleanup remove this `allow` once we populate a variant below
-        #[allow(clippy::match_single_binding)]
         match zone.zone_type.kind() {
-            // <https://github.com/oxidecomputer/omicron/issues/6404>
-            // ZoneKind::CockroachDb => todo!("check cluster status in inventory"),
+            ZoneKind::CockroachDb => {
+                debug!(self.log, "Checking if Cockroach node can shut down");
+                // We must hear from all nodes
+                let all_statuses = &self.inventory.cockroach_status;
+                if all_statuses.len() < COCKROACHDB_REDUNDANCY {
+                    debug!(self.log, "Not enough nodes");
+                    return false;
+                }
+
+                // All nodes must report: "We have the necessary redundancy, and
+                // have observed no underreplicated ranges".
+                for (_node_id, status) in all_statuses {
```

Comment:
Could include the node ID in the log messages (and later the more structured "why we can't update" results), so we know which node is blocking update?

I wonder if it's worthwhile to gather all the ways from all the nodes we might be blocking an update, instead of only logging the first one?

Neither of these is at all urgent, and both would be fine to defer to #8284

Reply:
Went ahead and updated the logs in 55e30a32789b3e11e2c21a42c2ed3b5bd9ebefe9 - but I'm going to defer to #8284 to make this structured, since that seems dependent on @plotnick 's work.